content/blog/2018-08-21-birds-radolfzell.md

```diff
@@ -0,0 +1,520 @@
+
+Now that we have the occurrence data, let’s plot it to see whether
+trimming is required.
+
+``` r
+crs <- sf::st_crs(landkreis_konstanz)
+
+birds_sf <- sf::st_as_sf(birds,
+                         coords = c("lng", "lat"), 
+                         crs = crs)
+library("ggplot2")
+ggplot() +
+  geom_sf(data = landkreis_konstanz) +
+  geom_sf(data = birds_sf) +
+  theme(legend.position = "bottom") +
+  hrbrthemes::theme_ipsum()
+```
+
+![](/img/blog-images/2018-08-21-birds-radolfzell/unnamed-chunk-4-1.png)
+
+Yes, trimming is required! It’d have been too bad not to learn how to do
+it, anyway.
+
+``` r
+in_indices <- sf::st_within(birds_sf, landkreis_konstanz)
+```
+
+    ## although coordinates are longitude/latitude, st_within assumes that they are planar
+
+``` r
+trimmed_birds <- dplyr::filter(birds_sf,
+                               lengths(in_indices) > 0)
+
+summarized_birds <- trimmed_birds %>%
+  dplyr::group_by(locName) %>%
+  dplyr::summarise(n = n())
+
+ggplot() +
+  geom_sf(data = landkreis_konstanz) +
+  geom_sf(data = summarized_birds,
+          aes(size = n), show.legend = "point") +
+  hrbrthemes::theme_ipsum()
+```
+
+![](/img/blog-images/2018-08-21-birds-radolfzell/unnamed-chunk-5-1.png)
+
+We got 47 observations (`nrow(trimmed_birds)`) of 47 species
+(`length(unique(trimmed_birds$comName))`), over 2 places
+(`length(unique(trimmed_birds$locName))`) during 4 observation sessions.
+Hopefully merely an appetizer to what we can hope to get from using the
+full eBird dataset in the next section…
+
+Before then, let’s just add the MPI on the map.
+
+``` r
+mpi <- opencage::opencage_forward("Am Obstberg 1 78315 Radolfzell", limit = 1)$results
+coords <- data.frame(lon = mpi$geometry.lng,
+                     lat = mpi$geometry.lat)
+crs <- sf::st_crs(landkreis_konstanz)
+
+mpi_sf <- sf::st_as_sf(coords,
+                       coords = c("lon", "lat"), 
+                      crs = crs)
+
+ggplot() +
+  geom_sf(data = landkreis_konstanz) +
+  geom_sf(data = summarized_birds,
+          aes(size = n), show.legend = "point") +
+  geom_sf(data = mpi_sf,
+          shape = 2) +
+  hrbrthemes::theme_ipsum()
+```
+
+![](/img/blog-images/2018-08-21-birds-radolfzell/unnamed-chunk-6-1.png)
+
+Note that the initial query could have been made with `spocc` which
+would have helped using the rOpenSci occurrence suite.
+
+``` r
+birds2 <- spocc::occ(from = "ebird",
+                     ebirdopts = list(method = "ebirdgeo",
+                                      species = NULL,
+                                      lng = center["x.x1"],
+                                      lat = center["y.y1"],
+                                      back = 30,
+                                      dist = as.numeric(
+                                        units::set_units(dist, "km"))))
+                     
+mapr::map_leaflet(birds2)
+```
+
+    ## no. taxa > 9, using single color - consider passing in colors
```

Comment:
you could set `warnings=FALSE` and possibly `messages=FALSE` for the knitr chunks to avoid thsese